### PR TITLE
binutils, lld: Use update-alternatives for `ld`

### LINKED ISSUE #8191: [Bug]: clang: ld: command not found
### Problem description

Spun off from #8183.

The `clang` package does not install `$PREFIX/bin/ld`.

Currently `bin/ld` is provided by the `binutils` package, which installs `ld.bfd` under the name of `ld`. But `clang` package does not depend on `binutils`. Note that Clang/LLVM on Termux is configured to use `lld` as the default linker, so it should not have any dependency on `binutils`.

I suggest that `bin/ld` be a symbolic link managed by update-alternatives, pointing either `bin/lld` or `bin/ld.bfd` (renamed from `bin/ld` in `binutils`).

### What steps will reproduce the bug?

```
pkg install clang
ld
```
produces

```
bash: ld: command not found
```

### What is the expected behavior?

```
ld: error: no input files
```

### System information

Tested on Android 8.1.0 aarch64.

### PR DESCRIPTION
**Pull request by xtkoba**
Closes #8191.

### CODE REVIEW

File: packages/binutils/build.sh [hunk 1]
```diff
@@ -41,8 +41,30 @@ termux_step_post_make_install() {
 	cd $TERMUX_PREFIX/bin
 	# Setup symlinks as these are used when building, so used by
 	# system setup in e.g. python, perl and libtool:
-	for b in ar ld nm objdump ranlib readelf strip; do
+	for b in ar nm objdump ranlib readelf strip; do
 		ln -s -f $b $TERMUX_HOST_PLATFORM-$b
 	done
+	ln -sf ld.bfd $TERMUX_HOST_PLATFORM-ld
```
Comment (buttaface) [inline packages/binutils/build.sh:47]: Can you move this to the lld package and have this `$TERMUX_HOST_PLATFORM-ld` symbolic link redirect to lld instead? The NDK only uses lld now, so we should do the same.

**Review by buttaface — APPROVED** [review]
Good move, I did not consider this when switching over to lld with NDK 23.

File: packages/binutils/build.sh (hunk 1 above)
Comment (buttaface) [reply packages/binutils/build.sh:47]: Ping, if you make this change, we can get this in.

Comment (buttaface) [reply packages/binutils/build.sh:47]: Ping again, if you update this, we can get it in, @xtkoba.

**Comment by licy183** [comment]
How about adding packages like `ld-is-lld`, `ld-is-ld-bfd` and `ld-is-ld-gold` as how ubuntu 20.04 deals with the `python` version? See https://packages.ubuntu.com/source/focal/what-is-python.

**Comment by buttaface** [comment]
Don't think it's worth it, as you can always do so manually, but not against it.

**Pull request closed without merging**